ocrd_webapi/database.py

```diff
@@ -288,3 +289,30 @@ async def get_workflow_job_state(job_id) -> Union[str, None]:
 @call_sync
 async def sync_get_workflow_job_state(job_id) -> Union[str, None]:
     return await get_workflow_job_state(job_id)
+
+
+async def get_user(email: str) -> Union[UserAccountDB, None]:
+    return await UserAccountDB.find_one(UserAccountDB.email == email)
+
+
+@call_sync
+async def sync_get_user(email: str) -> Union[UserAccountDB, None]:
```

Comment:
This method is not used anywhere.

Reply:
It's not because it's not needed here. But the repo already provides async and sync versions of all db methods. That's why this method is there.

Reply:
I will not include code which is not used. If we always say "in case it's needed in the future", we would then have unlimited cases to consider.

Reply:
> I will not include code which is not used.

From experience, I am telling you it's better to have these methods since the WebAPI is or pretends to be a library used or to be used by other projects. It provides certain flexibility to the implementers. There are also other DB methods (sync versions) not used directly in this repo itself but were required for simplified implementation.

Reply:
Then, we need to differentiate if we're writing a library or an "end-program". Do you expect other people to do something like `import ocrd-webapi-implementation`? I don't think so.

For me, this is an end-program. Other projects use this as a reference implementation. If they want to adapt some parts, they will fork this repo and make changes accordingly.

Reply:
> Then, we need to differentiate if we're writing a library or an "end-program".

Is the OCR-D core a library or an "end program"? The code in this repo will be transferred into the core as `ocrd_network` in the near future.

> Do you expect other people to do something like import ocrd-webapi-implementation?

Have you been able to check how the WebAPI is currently utilized in Operandi?

> For me, this is an end-program. Other projects use this as a reference implementation.

Not really... Yes, this repo started just as a reference implementation but now it is more than a reference implementation. And is in fact flexible enough to be used as a library. Letting the implementer decide if they want to reuse the routers as they are, override some of the endpoints of a specific router, or extend the router by adding more endpoints.

> If they want to adapt some parts, they will fork this repo and make changes accordingly.

Then what is the point of the `ocrd_network` package in the core at all?

I again feel like we are losing precious time with this debate here while there are more important topics that still need attention (i.e. https://github.com/OCR-D/core/discussions/1015).